astropy/modeling/functional_models.py

```diff
@@ -1315,6 +1315,75 @@ def __init__(self):
         raise ModelDefinitionError("Not implemented")
 
 
+class RampForBox1D(Fittable1DModel):
```

Comment:
Should we just call it `Ramp1D`? This does mean initializing it with an amplitude that is box amplitude / box width, but I think it means also that the function is more generally useful.

Although possibly even better would be to combine a new `Clip` model with `Linear1D`...